### ISSUE TITLE
ENH: don't raise in integer division by zero

### ISSUE THREAD

**Issue by jbrockmendel** [apache/arrow#33843]
### Describe the enhancement requested

`pc.divide_checked` and `pc.divide` both raise ArrowInvalid on division by zero with integer dtypes.  For at least the non-checked version it would be more useful to behave like floating dtypes:

```
>>> arr = pa.array([-1, 0, 1])
>>> pc.divide(arr, 0)
[...]
pyarrow.lib.ArrowInvalid: divide by zero

arr2 = arr.cast(pa.float64())
>>> pc.divide(arr2, 0)
<pyarrow.lib.DoubleArray object at 0x1293cd300>
[
  -inf,
  nan,
  inf
]
```

cc @jorisvandenbossche 

BTW I'm a big fan of the overflow-checking!

### Component(s)

Python

**Comment by jorisvandenbossche** [comment]
> If you want this with Arrow, I think the only workaround right now is to explicitly cast one of both operands to float to trigger standard floating point division

That's also what is done in the R dplyr bindings:

https://github.com/apache/arrow/blob/22e3bed4d1820f763ff04595d4baf171e7ecae2d/r/R/expression.R#L183-L187

**Comment by jbrockmendel** [comment]
Is casting to floats before operating going to be fragile for large integers?

**Comment by jorisvandenbossche** [comment]
Yes, but there might not be a way around that? Looking at numpy, they only seem to implement the core true_divide ufunc loops for floats, so I assume that numpy also casts to float before doing the operation.

**Comment by thisisnic** [comment]
This issue has been marked as stale because it has had no activity in the past 2 years. Please remove the stale label or comment below, or this issue will be closed in 14 days.
